src/rgw/rgw_rest_log.cc

```diff
@@ -552,7 +607,19 @@ void RGWOp_BILog_Delete::execute(optional_yield y) {
       return;
     }
   }
-  op_ret = store->svc()->bilog_rados->log_trim(bucket_info, shard_id, start_marker, end_marker);
+
+  const auto& logs = bucket_info.layout.logs;
+  auto log_layout = std::reference_wrapper{logs.back()};
+  if (gen) {
+    auto i = std::find_if(logs.begin(), logs.end(), matches_gen(*gen));
+    if (i == logs.end()) {
+      cerr << "ERROR: no log layout with gen=" << *gen << std::endl;
+      return ENOENT;
```

Comment:
in ops, errors should be logged with `dpp_dout()` and errors assigned to `op_ret`